src/mesh/meshblock.cpp

```diff
@@ -155,6 +155,7 @@ MeshBlock::MeshBlock(int igid, int ilid, LogicalLocation iloc, RegionSize input_
   if (SELF_GRAVITY_ENABLED) {
     // if (this->grav_block)
     pgrav = new Gravity(this, pin);
+    pgrav->EnrollSource(phydro->u, IDN);
```

Comment:
If you agree with the comment on the style of `enrolled_src_`, this can be enrolling `den` as a shallow slice of `u`.